src/o365/spo/commands/tenant/tenant-recyclebinitem-restore.ts

```diff
@@ -0,0 +1,149 @@
+      spoAdminUrl = adminUrl;
+      return this.restoreDeletedSite(args.options.url, spoAdminUrl, cmd);
+    })
+    .then((response: RestSpoOperation): void => {
+      if (!response.HasTimedout && response.IsComplete) {
+        if (this.verbose) {
+          cmd.log('Site Collection restored');
+        }
+      }
+      else if (wait) {
+        this.waitForRestoreDeletedSite(args.options.url, spoAdminUrl, cmd, response);
+      }
+      else {
+        cmd.log('Site Collection has not been restored');
+      }
+
+      cb();
+    }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  public cancel(): CommandCancel {
+    return (): void => {
+      if (this.timeout) {
+        clearTimeout(this.timeout);
+      }
+    }
+  }
+
+  private waitForRestoreDeletedSite(siteToRestoreUrl: string, spoAdminUrl: string, cmd: CommandInstance, response: RestSpoOperation): void {
+    this.timeout = setTimeout(() => {
+      this.restoreDeletedSite(siteToRestoreUrl, spoAdminUrl, cmd)
+      .then((respRetry: RestSpoOperation) => {
+        this.waitForRestoreDeletedSite(siteToRestoreUrl, spoAdminUrl, cmd, respRetry);
+      })
+      .catch((err) => {
+        cmd.log('Site Collection has not been restored');
+      });
+    }, response.PollingInterval);
+  }
+
+  private restoreDeletedSite(siteToRestoreUrl: string, spoAdminUrl: string, cmd: CommandInstance): Promise<RestSpoOperation> {
+    if (this.verbose) {
+      cmd.log('Restoring the deleted Site Collection');
+    }
+
+    if (this.debug) {
+      cmd.log(`Submitting POST request to ${spoAdminUrl}/_api/Microsoft.Online.SharePoint.TenantAdministration.Tenant/RestoreDeletedSite`);
+      cmd.log(`siteUrl : ${siteToRestoreUrl}`);
+    }
+    const requestOptions: any = {
+      url: `${spoAdminUrl}/_api/Microsoft.Online.SharePoint.TenantAdministration.Tenant/RestoreDeletedSite`,
+      headers: {
+        accept: 'application/json;odata=nometadata',
+        contenttype: 'application/json;odata=nometadata',
+      },
+      body: {
+        siteUrl: siteToRestoreUrl
+      },
+      json: true
+    };
+
+    return request.post(requestOptions);
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-u, --url <url>',
+        description: 'URL of the Site Collection to restore'
+      },
+      {
+        option: '--wait',
+        description: 'Wait for the Site Collection to be restored before completing the command'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+    // return options;
+  }
+
+  public commandHelp(args: CommandArgs, log: (help: string) => void): void {
+    const chalk = vorpal.chalk;
+    log(vorpal.find(commands.TENANT_RECYCLEBINITEM_RESTORE).helpInformation());
+    log(
+      ` ${chalk.yellow('Important:')} you have to be at least a SharePoint Administrator (not just an owner) to use this command.
```

Comment:
We have a standard statement for this that we use in other tenant-admin-level commands:

```text
${chalk.yellow('Important:')} to use this command you have to have permissions to access
      the tenant admin site.
```